### PR TITLE
ddl: make concurrent ingest job queue instead of fallback

### LINKED ISSUE #47758: [global sort] can not add index with global sort concurrency
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. enable global sort
2. add index for 2 tables concurrenty

### 2. What did you expect to see? (Required)
1. all running ddl jobs can use global sort add index
### 3. What did you see instead (Required)
1. only 1 ddl job can use ingest model and global sort
2. another add index job using txn merge 
### 4. What is your TiDB version? (Required)
```sql
+-----------------------------------------------------------+
| tidb_version()                                            |
+-----------------------------------------------------------+
| Release Version: v7.5.0-alpha                             |
| Edition: Community                                        |
| Git Commit Hash: 79db4ec58be3856995d31013e20f3faa57e3a3ac |
| Git Branch: heads/refs/tags/v7.5.0-alpha                  |
| UTC Build Time: 2023-10-16 11:42:39                       |
| GoVersion: go1.21.3                                       |
| Race Enabled: false                                       |
| Check Table Before Drop: false                            |
| Store: tikv                                               |
+-----------------------------------------------------------+

```

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #47758

Problem Summary:

1. only 1 ddl job can use ingest model and global sort.
1. another add index job using txn merge.

### What is changed and how it works?

- This PR fixes problem 2 by making the concurrent adding index job queue.
- Remove useless code.
- Refactor improper usage of system variable.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  Add `time.Sleep(20*time.Second)` to the first phase of `onCreateIndex()`.
  Session 1:
  ```
  mysql> alter table t add index i1 (a);
  Query OK, 0 rows affected (27.02 sec)
  ```
  Session 2:
  ```
  mysql> alter table t add index i2 (b);
  Query OK, 0 rows affected (46.31 sec)
  mysql> admin show ddl jobs 2;
  +--------+---------+------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
  | JOB_ID | DB_NAME | TABLE_NAME | JOB_TYPE               | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | CREATE_TIME         | START_TIME          | END_TIME            | STATE  |
  +--------+---------+------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
  |    110 | test    | t          | add index /* ingest */ | public       |         2 |      104 |         3 | 2023-11-16 16:47:15 | 2023-11-16 16:47:38 | 2023-11-16 16:48:01 | synced |
  |    109 | test    | t          | add index /* ingest */ | public       |         2 |      104 |         3 | 2023-11-16 16:47:14 | 2023-11-16 16:47:14 | 2023-11-16 16:47:38 | synced |
  +--------+---------+------------+------------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
  2 rows in set (0.00 sec)
  ```
  Both jobs can use ingest now. You can also check the log:
  ```
  [2023/11/16 16:47:15.427 +08:00] [INFO] [backend_mgr.go:79] ["ingest backfill worker is already in used by another DDL job"] [category=ddl-ingest] ["processing job ID"=109]
  ```
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/ddl_api.go [hunk 1]
```diff
@@ -7338,6 +7338,8 @@ func (d *ddl) createIndex(ctx sessionctx.Context, ti ast.Ident, keyType ast.Inde
 		Charset:    chs,
 		Collate:    coll,
 	}
+	job.ReorgMeta.IsDistReorg = variable.EnableDistTask.Load()
+	job.ReorgMeta.IsFastReorg = variable.EnableFastReorg.Load()
```
Comment (ywqzzy) [inline pkg/ddl/ddl_api.go:7342]: If enbaleDistTask == true and enableFastReorg == false, what will happen?

Comment (tangenta) [reply pkg/ddl/ddl_api.go:7342]:
```
mysql> set global tidb_enable_dist_task = 1;
Query OK, 0 rows affected (0.02 sec)

mysql> set global tidb_ddl_enable_fast_reorg = 0;
Query OK, 0 rows affected (0.03 sec)

mysql> alter table t add index i3(c);
Query OK, 0 rows affected (0.43 sec)

mysql> admin show ddl jobs 1;
+--------+---------+------------+---------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
| JOB_ID | DB_NAME | TABLE_NAME | JOB_TYPE            | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | CREATE_TIME         | START_TIME          | END_TIME            | STATE  |
+--------+---------+------------+---------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
|    111 | test    | t          | add index /* txn */ | public       |         2 |      104 |         3 | 2023-11-16 17:16:52 | 2023-11-16 17:16:52 | 2023-11-16 17:16:53 | synced |
+--------+---------+------------+---------------------+--------------+-----------+----------+-----------+---------------------+---------------------+---------------------+--------+
1 row in set (0.01 sec)
```

**Review by ywqzzy — APPROVED** [review]

File: errors.toml [hunk 2]
```diff
@@ -1343,7 +1343,7 @@ Unsupported clustered primary key type FLOAT/DOUBLE for TTL
 
 ["ddl:8200"]
 error = '''
-Unsupported shard_row_id_bits for table with primary key as row id
+Unsupported tidb_enable_dist_task setting. Please enable tidb_ddl_enable_fast_reorg to use distributed task execution.
```
Comment (Benjamin2037) [inline errors.toml:1346]: Why here use 
```suggestion
Unsupported tidb_enable_dist_task setting. To utilize distributed task execution, please enable tidb_ddl_enable_fast_reorg first.
```

File: pkg/ddl/multi_schema_change.go [hunk 3]
```diff
@@ -55,6 +65,16 @@ func (d *ddl) MultiSchemaChange(ctx sessionctx.Context, ti ast.Ident) error {
 	return d.callHookOnChanged(job, err)
 }
 
+func containsDistTaskSubJob(subJobs []*model.SubJob) bool {
+	for _, sub := range subJobs {
+		if sub.Type == model.ActionAddIndex ||
```
Comment (Benjamin2037) [inline pkg/ddl/multi_schema_change.go:70]: Once there is add index or add pk then we make it distributed sub job? this means the multi schema change support part subjob be executed on dist framework?

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: pkg/ddl/multi_schema_change.go (hunk 3 above)
Comment (tangenta) [reply pkg/ddl/multi_schema_change.go:70]: Yes.

**Pull request merged**